R/VisualizeData.R

```diff
@@ -268,36 +273,41 @@ plotEmbedding <- function(
   .validInput(input = keepAxis, name = "keepAxis", valid = c("boolean"))
   .validInput(input = baseSize, name = "baseSize", valid = c("numeric"))
   .validInput(input = plotAs, name = "plotAs", valid = c("character", "null"))
+  .validInput(input = Shiny, name = "Shiny", valid = c("boolean"))
   .validInput(input = threads, name = "threads", valid = c("integer"))
   .validInput(input = logFile, name = "logFile", valid = c("character"))
-
+  
   .requirePackage("ggplot2", source = "cran")
-
+  
   .startLogging(logFile = logFile)
   .logThis(mget(names(formals()),sys.frame(sys.nframe())), "Input-Parameters", logFile=logFile)
-
+  
   ##############################
   # Get Embedding
   ##############################
-  .logMessage("Getting UMAP Embedding", logFile = logFile)
-  df <- getEmbedding(ArchRProj, embedding = embedding, returnDF = TRUE)
+  .logMessage("Getting Embedding", logFile = logFile)
+  if(Shiny){
+     df <- embeddingDF
+  } else{
+    df <- getEmbedding(ArchRProj, embedding = embedding, returnDF = TRUE)
+  }
 
   if(!all(rownames(df) %in% ArchRProj$cellNames)){
     stop("Not all cells in embedding are present in ArchRProject!")
   }
-
   .logThis(df, name = "Embedding data.frame", logFile = logFile)
+  
   if(!is.null(sampleCells)){
     if(sampleCells < nrow(df)){
       if(!is.null(imputeWeights)){
-        stop("Cannot sampleCells with imputeWeights not equalt to NULL at this time!")
+        stop("Cannot sampleCells with imputeWeights not equal to NULL at this time!")
       }
       df <- df[sort(sample(seq_len(nrow(df)), sampleCells)), , drop = FALSE]
     }
   }
-
+  
   #Parameters
-  plotParams <- list(...)
```

Comment:
this `...` is actually essential. It allows for the use of passthrough arguments. I have reverted this deletion